Plugins/LoggerPlugin/LoggerPlugin.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import PackagePlugin
+import Foundation
+
+@main
+struct LoggerPlugin: BuildToolPlugin {
+  func createBuildCommands(context: PluginContext, target: Target) async throws -> [Command] {
+    let outputDirectory = context.pluginWorkDirectory.appending("GeneratedSources")
+    guard let target = target as? SourceModuleTarget else {
+      Diagnostics.error("Attempted to use `LoggerPlugin` on an unsupported module target")
+      return []
+    }
+    
+    try FileManager.default.createDirectory(atPath: outputDirectory.string, withIntermediateDirectories: true)
+    let source = """
+    import Foundation
+    import os.log
+    
+    public extension Logger {
```

Comment:
The `public` accessor here should be removed

Reply:
No niche use where we want to call logger from another module?

Reply:
If we were to expose a public logger for the module we would want its name to be different from `module` anyways so either way we want this `module` one to be internal only